### PR TITLE
[core] Sort props ASC

### PR DESCRIPTION
**Pull request by oliviertassinari**
One step closer to solve #1631 and https://github.com/mui-org/material-ui/issues/19706.

### CODE REVIEW

File: lib/src/_shared/hooks/useIsLandscape.tsx [hunk 1]
```diff
@@ -29,12 +29,15 @@ export function useIsLandscape(
     getOrientation()
   );
 
-  const eventHandler = React.useCallback(() => setOrientation(getOrientation()), []);
```
Comment (oliviertassinari) [inline lib/src/_shared/hooks/useIsLandscape.tsx:32]: Simplification

**Comment by dmtrKovalenko** [comment]
Do you make all this sortings manually? If so – we need some automate solution for that

**Comment by oliviertassinari** [comment]
We have a constraint on sorting the prop types, otherwise eslint crashes. It would be cool if we can extended this static analysis.

File: lib/src/views/Calendar/SlideTransition.tsx [hunk 2]
```diff
@@ -69,10 +69,10 @@ export const useStyles = makeStyles(
 
 export const SlideTransition: React.SFC<SlideTransitionProps> = ({
   children,
-  transKey,
+  className,
   reduceAnimations,
   slideDirection,
-  className = undefined,
```
Comment (oliviertassinari) [inline lib/src/views/Calendar/SlideTransition.tsx:75]: Remove dead code.

**Review by dmtrKovalenko — APPROVED** [review]

**Pull request merged**